pkg/storage/common/common.go

```diff
@@ -457,6 +457,81 @@ func PruneImageManifestsFromIndex(imgStore storageTypes.ImageStore, repo string,
 	return prunedManifests, nil
 }
 
+func blobUsedInImageManifest(imgStore storageTypes.ImageStore, repo string,
+	bdigest, mdigest godigest.Digest, log zerolog.Logger,
+) (bool, error) {
+	if bdigest == mdigest {
+		return true, nil
+	}
+
+	manifestContent, err := GetImageManifest(imgStore, repo, mdigest, log)
+	if err != nil {
+		log.Error().Err(err).Str("repo", repo).Str("digest", mdigest.String()).
+			Msg("gc: failed to read manifest image")
+
+		return false, err
+	}
+
+	if bdigest == manifestContent.Config.Digest {
+		return true, nil
+	}
+
+	for _, layer := range manifestContent.Layers {
+		if bdigest == layer.Digest {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func blobUsedInImageIndex(imgStore storageTypes.ImageStore, repo string,
+	digest godigest.Digest, index ispec.Index, log zerolog.Logger,
+) (bool, error) {
+	for _, desc := range index.Manifests {
+		var found bool
+
+		switch desc.MediaType {
+		case ispec.MediaTypeImageIndex:
+			/* this branch is not needed, because every manifests in index is already checked
+			when this one is hit, all manifests are referenced in index.json */
+			indexImage, err := GetImageIndex(imgStore, repo, desc.Digest, log)
+			if err != nil {
+				log.Error().Err(err).Str("repository", repo).Str("digest", desc.Digest.String()).
+					Msg("failed to read multiarch(index) image")
+
+				return false, err
+			}
+
+			found, _ = blobUsedInImageIndex(imgStore, repo, digest, indexImage, log)
+		case ispec.MediaTypeImageManifest:
+			found, _ = blobUsedInImageManifest(imgStore, repo, digest, desc.Digest, log)
+		}
+
+		if found {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func BlobInUse(imgStore storageTypes.ImageStore, repo string,
```

Comment:
Function name should contain a verb.
Same applies for the other functions defined in this file.